### PR TITLE
Structure_Engine: Compliance Update

### LINKED ISSUE #1523: Structure_Engine - Compliance issues
This report has been generated as part of a close out for the 3.1 milestone compliance and outlines all of the compliance issues currently highlighted by the CI team. @IsakNaslundBh  will be undertaking this issue to resolve all compliance issues which do not impact scripts or versioning, with the remainder being moved to the 3.2 milestone for further discussion and resolution.

The following compliance issues have been found for this project. Please see the checklist below.
##### Create file  must match the create method(s) return type. Either part of the path or the filename must be an exact match of the return type of the method. If the file name does not match the return type file name must match the method name.
 - [x] in BHoM_Engine\Structure_Engine\Create\LinkConstraints.cs at line 35, column 37 
 - [x] in BHoM_Engine\Structure_Engine\Create\LinkConstraints.cs at line 57, column 37 
 - [x] in BHoM_Engine\Structure_Engine\Create\LinkConstraints.cs at line 79, column 37 
 - [x] in BHoM_Engine\Structure_Engine\Create\LinkConstraints.cs at line 95, column 37 
 - [x] in BHoM_Engine\Structure_Engine\Create\LinkConstraints.cs at line 109, column 37 
 - [x] in BHoM_Engine\Structure_Engine\Create\LinkConstraints.cs at line 123, column 37 
 - [x] in BHoM_Engine\Structure_Engine\Create\LinkConstraints.cs at line 137, column 37 
 - [x] in BHoM_Engine\Structure_Engine\Create\LinkConstraints.cs at line 150, column 37 
 - [x] in BHoM_Engine\Structure_Engine\Create\LinkConstraints.cs at line 163, column 37 
 - [x] in BHoM_Engine\Structure_Engine\Create\LinkConstraints.cs at line 176, column 37 
 - [x] in BHoM_Engine\Structure_Engine\Create\LinkConstraints.cs at line 190, column 37 
 - [x] in BHoM_Engine\Structure_Engine\Create\LinkConstraints.cs at line 204, column 37 
 - [x] in BHoM_Engine\Structure_Engine\Create\LinkConstraints.cs at line 218, column 37 
 - [x] in BHoM_Engine\Structure_Engine\Create\LinkConstraints.cs at line 230, column 37 
 - [x] in BHoM_Engine\Structure_Engine\Create\LinkConstraints.cs at line 242, column 37 
 - [x] in BHoM_Engine\Structure_Engine\Create\Load.cs at line 346, column 40 
 - [x] in BHoM_Engine\Structure_Engine\Create\Load.cs at line 366, column 40 
 - [x] in BHoM_Engine\Structure_Engine\Create\Load.cs at line 373, column 40 
 - [x] in BHoM_Engine\Structure_Engine\Create\Load.cs at line 392, column 40 
 - [x] in BHoM_Engine\Structure_Engine\Create\Load.cs at line 442, column 36 
 - [x] in BHoM_Engine\Structure_Engine\Create\Load.cs at line 461, column 36 
 - [x] in BHoM_Engine\Structure_Engine\Create\Loadcombination.cs at line 36, column 38 
 - [x] in BHoM_Engine\Structure_Engine\Create\Loadcombination.cs at line 43, column 38 
 - [x] in BHoM_Engine\Structure_Engine\Create\MaterialFragment.cs at line 48, column 32 
 - [x] in BHoM_Engine\Structure_Engine\Create\MaterialFragment.cs at line 74, column 31 
 - [x] in BHoM_Engine\Structure_Engine\Create\MaterialFragment.cs at line 102, column 28 
 - [x] in BHoM_Engine\Structure_Engine\Create\MaterialFragment.cs at line 129, column 29 
 - [x] in BHoM_Engine\Structure_Engine\Create\MeshResult.cs at line 41, column 36 
 - [x] in BHoM_Engine\Structure_Engine\Create\NewElement0D.cs at line 36, column 33 
 - [x] in BHoM_Engine\Structure_Engine\Create\NewElement1D.cs at line 37, column 33 
 - [x] in BHoM_Engine\Structure_Engine\Create\NewElement1D.cs at line 44, column 33 
 - [x] in BHoM_Engine\Structure_Engine\Create\NewInternalElement2D.cs at line 35, column 33 
 - [x] in BHoM_Engine\Structure_Engine\Create\SectionProperty.cs at line 47, column 42 

##### Input parameter should start with a lowercase letter
 - [x] in BHoM_Engine\Structure_Engine\Create\MaterialFragment.cs at line 48, column 62 
 - [x] in BHoM_Engine\Structure_Engine\Create\MaterialFragment.cs at line 74, column 60 
 - [x] in BHoM_Engine\Structure_Engine\Create\MaterialFragment.cs at line 102, column 54 
 - [x] in BHoM_Engine\Structure_Engine\Create\MaterialFragment.cs at line 129, column 56 
 - [x] in BHoM_Engine\Structure_Engine\Create\MaterialFragment.cs at line 129, column 76 
 - [x] in BHoM_Engine\Structure_Engine\Create\NewInternalElement2D.cs at line 35, column 65 
 - [x] in BHoM_Engine\Structure_Engine\Modify\SetInternalElements2D.cs at line 47, column 61 
 - [x] in BHoM_Engine\Structure_Engine\Modify\SetOutlineElements1D.cs at line 46, column 60 

##### Input attribute does not match any of the given parameters
 - [x] in BHoM_Engine\Structure_Engine\Create\IResultRequest.cs at line 44, column 15 

##### Method name must match file name
 - [x] in BHoM_Engine\Structure_Engine\Compute\Integrate.cs at line 174, column 29 
 - [x] in BHoM_Engine\Structure_Engine\Compute\ShearArea.cs at line 50, column 29 
 - [x] in BHoM_Engine\Structure_Engine\Modify\SetReinforcement.cs at line 45, column 38 
 - [x] in BHoM_Engine\Structure_Engine\Query\Edges.cs at line 58, column 35 
 - [x] in BHoM_Engine\Structure_Engine\Query\Edges.cs at line 73, column 35 
 - [x] in BHoM_Engine\Structure_Engine\Query\Edges.cs at line 83, column 35 
 - [x] in BHoM_Engine\Structure_Engine\Query\Edges.cs at line 95, column 35

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

Closes #1523 

Closes all compliance issues raised on Structure_Engine.

Some methods have been deprecated, as to be replaced by Auto create ones. Might have been a bit of a scope creep, but wanted to get everything in order. Should be no other functional change.

Also, had already started added description for Bar create, so got that included.

### CODE REVIEW

File: Structure_Engine/Create/Loads/Load.cs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+
+
+
+        /***************************************************/
+
+
+        /***************************************************/
+
+
```
Comment (kThorsager) [inline Structure_Engine/Create/Loads/Load.cs]: Compliant?

File: Structure_Engine/Create/MaterialFragments/Aluminium.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            {
+                Name = name,
+                Density = density,
+                YoungsModulus = youngsModulus,
+                PoissonsRatio = poissonsRatio,
+                ThermalExpansionCoeff = thermalExpansionCoeff,
+                DampingRatio = dampingRatio,
+                EmbodiedCarbon = embodiedCarbon,
+            };
+        }
```
Comment (kThorsager) [inline Structure_Engine/Create/MaterialFragments/Aluminium.cs]: Why are there no default values in the oM which would make this redundant

File: Structure_Engine/Create/MaterialFragments/Aluminium.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        /***************************************************/
+
+        [Description("Creates a structural Aluminium material fragment to be used on analytical structural elements, or as a fragment of the physical material.")]
+        [InputFromProperty("youngsModulus")]
+        [InputFromProperty("poissonsRatio")]
+        [InputFromProperty("thermalExpansionCoeff")]
+        [InputFromProperty("density")]
+        [InputFromProperty("dampingRatio")]
+        [InputFromProperty("embodiedCarbon")]
+        [Output("aluminium", "The created tructural Aluminium material fragment.")]
```
Comment (kThorsager) [inline Structure_Engine/Create/MaterialFragments/Aluminium.cs]:
```suggestion
        [Output("aluminium", "The created structural Aluminium material fragment.")]
```

File: Structure_Engine/Create/MaterialFragments/Concrete.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        /***************************************************/
+
+        [Description("Creates a structural Concrete material fragment to be used on analytical structural elements, or as a fragment of the physical material.")]
+        [InputFromProperty("youngsModulus")]
+        [InputFromProperty("poissonsRatio")]
+        [InputFromProperty("thermalExpansionCoeff")]
+        [InputFromProperty("density")]
+        [InputFromProperty("dampingRatio")]
+        [InputFromProperty("embodiedCarbon")]
+        [Output("aluminium", "The created tructural Concrete material fragment.")]
```
Comment (kThorsager) [inline Structure_Engine/Create/MaterialFragments/Concrete.cs]:
```suggestion
        [Output("concrete", "The created structural Concrete material fragment.")]
```

File: Structure_Engine/Create/MaterialFragments/Steel.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        /***************************************************/
+
+        [Description("Creates a structural Steel material fragment to be used on analytical structural elements, or as a fragment of the physical material.")]
+        [InputFromProperty("youngsModulus")]
+        [InputFromProperty("poissonsRatio")]
+        [InputFromProperty("thermalExpansionCoeff")]
+        [InputFromProperty("density")]
+        [InputFromProperty("dampingRatio")]
+        [InputFromProperty("embodiedCarbon")]
+        [Output("aluminium", "The created tructural Steel material fragment.")]
```
Comment (kThorsager) [inline Structure_Engine/Create/MaterialFragments/Steel.cs]:
```suggestion
        [Output("steel", "The created structural Steel material fragment.")]
```

File: Structure_Engine/Create/MaterialFragments/Timber.cs [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+        [Description("Creates a structural Timber material fragment to be used on analytical structural elements, or as a fragment of the physical material.")]
+        [InputFromProperty("youngsModulus")]
+        [InputFromProperty("poissonsRatio")]
+        [InputFromProperty("shearModulus")]
+        [InputFromProperty("thermalExpansionCoeff")]
+        [InputFromProperty("density")]
+        [InputFromProperty("dampingRatio")]
+        [InputFromProperty("embodiedCarbon")]
+        [Output("aluminium", "The created tructural Timber material fragment.")]
```
Comment (kThorsager) [inline Structure_Engine/Create/MaterialFragments/Timber.cs]:
```suggestion
        [Output("timber", "The created structural Timber material fragment.")]
```

File: Structure_Engine/Create/MaterialFragments/Aluminium.cs (hunk 2 above)
Comment (IsakNaslundBh) [reply Structure_Engine/Create/MaterialFragments/Aluminium.cs]: Get your point here, and in general agree, but, it felt wrong to have default values in the oM, but less wrong in the create method. Sort of, defaulting a method feels less set in stone than defaulting a full object. Not sure how solid that argument is, but that is why.
Also, opted to keep these from ongoing discussion in https://github.com/BHoM/BHoM_Engine/pull/1474 , where we might want to switch where default values for things like embodied carbon is being pulled from.

File: Structure_Engine/Create/Loads/Load.cs (hunk 1 above)
Comment (IsakNaslundBh) [reply Structure_Engine/Create/Loads/Load.cs]: Nope, thanks

**Pull request merged**